tree/dataframe/inc/ROOT/RDF/GraphNode.hxx

```diff
@@ -128,6 +109,50 @@ public:
       }
       fShape = "box";
    }
+
+public:
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Creates a node with a name
+   GraphNode(std::string_view name, unsigned int id, ENodeType t) : fID(id), fName(name)
+   {
+      switch (t) {
+      case ENodeType::kAction: SetAction(/*hasRun=*/false); break;
+      case ENodeType::kDefine: SetDefine(); break;
+      case ENodeType::kFilter: SetFilter(); break;
+      case ENodeType::kRange: SetRange(); break;
+      case ENodeType::kRoot: SetRoot(); break;
+      case ENodeType::kUsedAction: SetAction(/*hasRun=*/true); break;
+      };
+   }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Appends a node on the head of the current node
+   void SetPrevNode(const std::shared_ptr<GraphNode> &node) { fPrevNode = node; }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Adds the column defined up to the node
+   void AddDefinedColumns(const std::vector<std::string> &columns) { fDefinedColumns = columns; }
+
+   std::string GetColor() const { return fColor; }
+   unsigned int GetID() const { return fID; }
+   std::string GetName() const { return fName; }
+   std::string GetShape() const { return fShape; }
+   GraphNode *GetPrevNode() const { return fPrevNode.get(); }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Gets the column defined up to the node
+   std::vector<std::string> GetDefinedColumns() { return fDefinedColumns; }
+
+   bool IsExplored() const { return fIsExplored; }
+   bool IsNew() const { return fIsNew; }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief Allows to stop the graph traversal when an explored node is encountered
+   void SetExplored() { fIsExplored = true; }
+
+   ////////////////////////////////////////////////////////////////////////////
+   /// \brief The node is considered just created
+   void SetIsNew(bool isNew) { fIsNew = isNew; }
```

Comment:
`SetIsNew` is only used in `src/RDFGraphUtils.cxx` to set it to `false`. Therefore, this method can be simplified (and in this way it would be more consistent with `SetExplored`).

Reply:
done